tests/mock_network/mod.rs

```diff
@@ -645,3 +607,50 @@ fn node_pause_and_resume() {
     poll_and_resend(&mut nodes);
     verify_invariant_for_all_nodes(&network, &mut nodes);
 }
+
+#[test]
+fn carry_out_parsec_pruning() {
+    let init_network_size = 7;
+    let network = Network::new(
+        NetworkParams {
+            elder_size: LOWERED_ELDER_SIZE,
+            safe_section_size: LOWERED_ELDER_SIZE,
+        },
+        None,
+    );
+    let mut nodes = create_connected_nodes(&network, init_network_size);
+    poll_and_resend(&mut nodes);
+
+    let mut parsec_versions = BTreeMap::new();
+
+    for node in nodes.iter_mut() {
+        let id = node.name();
+        let _ = node.inner.elder_state_mut().map(|state| {
+            let _ = parsec_versions.insert(id, state.parsec_last_version());
+            state.vote_for_prune()
```

Comment:
Should that be how this test happen, that we vote_prune explicitly?
How does this test the mechanism that trigger the vote_prune?

Reply:
I think with the introduction of the reset parsec during churn, here we can only dispatch a bunch of user-data to bump up the graph size?

Reply:
Yes, or let it gossip.